stan/math/rev/core/vari.hpp

```diff
@@ -182,8 +193,9 @@ using vari = vari_value<double>;
  * derivative with respect to the root of the derivative tree.
  *
  */
-template <typename T>
-class vari_value<T, require_eigen_dense_base_t<T>> : public vari_base {
+template <typename T, typename StrideType>
+class vari_value<T, StrideType, require_eigen_dense_base_t<T>>
```

Comment:
I don't think adding a stride template is the best option here. I would prefere to have `T` be the actual expression in the `vari_value`. You can use some metaprogramming to convert plain types to maps. This would also allow adding triangular views and transpositions.

Reply:
Yeah I agree. I'll play around with this today

Reply:
How do you feel about this version with a [`vari_view`](https://github.com/stan-dev/math/compare/develop...feature/vari-view-eigen#diff-130c5a75cc427d7d41715e9fca8281f4R177) that's used in the `var_value`. Though I'm having a major problem where in the [var_value test](https://github.com/stan-dev/math/compare/develop...feature/vari-view-eigen#diff-2b0d4604da2786aba9769c587db655c5R169) When I print out the result of `A.block()` and it's rows and columns it matches up to `A_var.block()`. But when I check these with the macro `EXPECT_MATRIX_FLOAT_EQ` it says that the returned `val_` has 0 rows and 0 columns?